sentinel-core/src/main/java/com/alibaba/csp/sentinel/util/function/Tuple2.java

```diff
@@ -31,11 +31,31 @@ public Tuple2(R1 r1, R2 r2) {
         return new Tuple2<R2, R1>(this.r2, this.r1);
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof Tuple2)) {
+            return false;
+        }
+        Tuple2 tuple2 = (Tuple2) o;
+
+        return this.r1.equals(tuple2.r1) && this.r2.equals(tuple2.r2);
```

Comment:
It might be better to use `Objects.equals(a, b)` here, as `r1` and `r2` might be null (refer the template in IDEA).

Reply:
done !  you are right,I find some equals() functions  in sentinel  exist the  same problems like my code

Reply:
thank you very much for your help!!

Reply:
> done! you are right, I find some equals() functions in sentinel exist the same problems like my code

Feel free to submit PRs to fix it :)